clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -6482,7 +6475,11 @@ llvm::DILocation *CodeGenFunction::SanitizerAnnotateDebugInfo(
     SanitizerHandler Handler) {
   llvm::DILocation *CheckDebugLoc = Builder.getCurrentDebugLocation();
   auto *DI = getDebugInfo();
-  if (!DI)
+  if (!DI || !CheckDebugLoc)
+    return CheckDebugLoc;
```

Comment:
Do all the extra early-exits speed up or slow down the codegen?

Reply:
The performance tradeoff depends on how often the check fails (if CheckDebugLoc is rarely null, then adding the check here may slow down codegen overall).

Reply:
We need to check it somewhere?